iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -646,6 +918,58 @@ PortManager::acquireSubscriberPortData(const capro::ServiceDescription& service,
     return maybeSubscriberPortData;
 }
 
+cxx::expected<popo::ClientPortData*, PortPoolError>
+PortManager::acquireClientPortData(const capro::ServiceDescription& service,
+                                   const popo::ClientOptions& clientOptions,
+                                   const RuntimeName_t& runtimeName,
+                                   mepoo::MemoryManager* const payloadDataSegmentMemoryManager,
+                                   const PortConfigInfo& portConfigInfo) noexcept
+{
+    // we can create a new port
+    return m_portPool
+        ->addClientPort(service, payloadDataSegmentMemoryManager, runtimeName, clientOptions, portConfigInfo.memoryInfo)
+        .and_then([this](auto clientPortData) {
+            /// @todo iox-#27 add to port introspection
+
+            // we do discovery here for trying to connect the client if offer on create is desired
+            popo::ClientPortRouDi clientPort(*clientPortData);
+            this->doDiscoveryForClientPort(clientPort);
+        });
+}
+
+cxx::expected<popo::ServerPortData*, PortPoolError>
+PortManager::acquireServerPortData(const capro::ServiceDescription& service,
+                                   const popo::ServerOptions& serverOptions,
+                                   const RuntimeName_t& runtimeName,
+                                   mepoo::MemoryManager* const payloadDataSegmentMemoryManager,
+                                   const PortConfigInfo& portConfigInfo) noexcept
+{
+    // it is not allowed to have two servers with the same ServiceDescription;
```

Comment:
Just an idea, can we fix #938 if we do check for the server to be deleted here. Sure do the same for the publisher where this bug was reported. Otherwise we would have the same problem with servers. And there it is more likely as two same servers is always not allowed. Something like 

```
    for (auto serverPortData : m_portPool->getServerPortDataList())
    {
        popo::ServerPortRouDi serverPort(*serverPortData);

        if (serverPort.toBeDestroyed())
        {
            destroyServerPort(serverPortData);
        }
    }
```

Reply:
@budrus for the publisher this could be done in `doesViolateCommunicationPolicy``similar to what I did with the server port. If you agree, I'll create a PR for that issue